hearts/hearts.py

```diff
@@ -261,22 +329,31 @@ def shot_the_moon(self):
         return {player: score == 26 for (player, score) in self.current_scores().items()}
 
     def final_scores(self):
-        scores = self.current_scores()
-        shoot_successes = self.shot_the_moon()
-
-        if any(shoot_successes.values()):
-            for (player, shoot_success) in shoot_successes.items():
-                scores[player] = 0 if shoot_success else 26
+        if self.is_over:
+            scores = self.current_scores()
+            shoot_successes = self.shot_the_moon()
+
+            if any(shoot_successes.values()):
+                for (player, shoot_success) in shoot_successes.items():
+                    scores[player] = 0 if shoot_success else 26
+            return scores
+        else:
+            return None
 
-        return scores
+    def is_over(self):
+        return len(self.tricks) == 13 and len(self.tricks[-1]) == 4
 
     def serialize(self):
         return {
             'players': self.players,
+            'direction': self.pass_to,
             'turn': self.players[self.turn_counter].username,
-            'hands': self.hands,
+            'hands': [hand.serialize() for hand in self.hands],
             'tricks': [trick.serialize() for trick in self.tricks],
-            'hearts': str(self.hearts_broken)
+            'hearts': str(self.hearts_broken),
+            'current_scores': self.current_scores(),
+            'final_scores': (self.final_scores() if self.is_over() else 'None'),
```

Comment:
Also, the serialized values should all be python values (e.g., `hearts_broken` is a boolean, don't turn it into a string)